### ISSUE TITLE
macOS 10.3.6 --> wmic: command not found

### ISSUE THREAD

**Issue by d3vv** [glinscott/fishtest#363]
p6tse:~ dew$ python fishtest/worker/worker.py 
Worker starting ...

sh: wmic: command not found
Traceback (most recent call last):
  File "fishtest/worker/worker.py", line 46, in setup_config_file
    mem = int(re.search(r'\d+', mem_str).group())
AttributeError: 'NoneType' object has no attribute 'group'
Worker version 67 connecting to http://tests.stockfishchess.org:80
Fetch task...
Worker version checked successfully in 0.144674s
Task requested in 0.286235s

**Comment by ppigazzini** [comment]
@snicolet Stéphane, if I'm not wrong you joined long time ago the dark side of the force. Can you help @d3vv to run the worker on macOS?

**Comment by d3vv** [comment]
And one more..
Under macOS gcc it is alias for clang.. Fishtest try to use gcc compilation with profile build by default..
And we have an error cause gcc pgo scheme is different against clang.. And we need to set "COMP=clang" for success..

**Comment by ppigazzini** [comment]
If macOS has the `free` command (with output is Linux like) the bugfix should be very easy. Try to modify by hand the code from this:

https://github.com/glinscott/fishtest/blob/abe0773af623699b2052057c5313e307595f1894/worker/worker.py#L39-L50

to this:

```python
 try: 
   if 'Linux' or 'Darwin' in platform.system(): 
```

**Comment by ppigazzini** [comment]
@d3vv the worker code base should have a refactoring to work seamless with Linux, Windows, msys2, MacOs...

**Comment by ppigazzini** [comment]
@d3vv I don't have macOS, so I need that you test the changes by hand before writing and pushing a PR.

Try this

```python
try: 
   if 'Linux' in platform.system(): 
     cmd = 'free' 
   elif 'Darwin' in platform.system():
     cmd = 'sysctl hw.memsize' 
   else: 
     cmd = 'wmic computersystem get TotalPhysicalMemory' 
   with os.popen(cmd) as proc:
   ...
```

**Comment by d3vv** [comment]
@ppigazzini thanx again... Personally I have no problems with hand modifications
I opened an issue for others

**Comment by d3vv** [comment]
Ahhh... okey I will test in a hour and report result

**Comment by d3vv** [comment]
@ppigazzini it seems all correct:

p6tse:~ dew$ diff fishtest/worker/worker.py fishtest/worker/worker.py.OLD 
42,43d41
<     elif 'Darwin' in platform.system():
<       cmd = 'sysctl hw.memsize'


p6tse:~ dew$ uname -a
Darwin p6tse 17.7.0 Darwin Kernel Version 17.7.0: Thu Dec 20 21:47:19 PST 2018; root:xnu-4570.71.22~1/RELEASE_X86_64 x86_64

My worker:

![screen shot 2019-02-01 at 14 57 18](https://user-images.githubusercontent.com/32866346/52121743-b5641980-2631-11e9-9fbd-8d4304f6ab21.png)

**Comment by d3vv** [comment]
But... p6tse:~ dew$ cat fishtest.cfg |grep max
max_memory = 0

**Comment by d3vv** [comment]
p6tse:~ dew$ sysctl hw.memsize
hw.memsize: 25769803776
p6tse:~ dew$

**Comment by d3vv** [comment]
for 24GB

**Comment by d3vv** [comment]
it seems string "hw.memsize: <value>" parse needed

**Comment by d3vv** [comment]
p6tse:~ dew$ sysctl hw.memsize|awk {'print $2'}
25769803776
p6tse:~ dew$

**Comment by ppigazzini** [comment]
@d3vv I think to have found a new bug in that code. Windows and macOS report memory in bytes, Linux in kbytes.
With python be sure to respect the indentation when writing the code in a script.

Open a python console (just start `python` in a terminal) and write this:
```
import os
import re
cmd = 'ysctl hw.memsize'
mem_str = str(os.popen(cmd).readlines())
print(mem_str)
mem = int(re.search(r'\d+', mem_str).group())
print(mem)

```

**Comment by ppigazzini** [comment]
@d3vv use [markdown](https://guides.github.com/features/mastering-markdown/) to write the comments, to make a long story short write code or console output the output in this way:
\`\`\`
\<put here code or console output\>
\`\`\`

and check the formatting result in the "Preview" tab.

**Comment by d3vv** [comment]
"cmd = 'ysctl" I changed to "sysctl" 

p6tse:~ dew$ python
Python 2.7.15 (default, Sep 12 2018, 13:32:25) 
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> import re
>>> cmd = 'sysctl hw.memsize'
>>> mem_str = str(os.popen(cmd).readlines())
>>> print(mem_str)
['hw.memsize: 25769803776\n']
>>> mem = int(re.search(r'\d+', mem_str).group())
>>> print(mem)
25769803776
>>>

another way:

p6tse:~ dew$ sysctl -n hw.memsize
25769803776

**Comment by d3vv** [comment]
@ppigazzini You are right
but with "-n" arg we can:

```
     -n      Show only variable values, not their names.  This option is useful for setting shell variables.  For
             instance, to save the pagesize in variable psize, use:

                   set psize=`sysctl -n hw.pagesize`
```

**Comment by ppigazzini** [comment]
@d3vv thank you for the debug. Before submitting a PR I need to verify if the Linux memory format (in Kb) is the correct one.

**Comment by d3vv** [comment]
@ppigazzini as for Linux:

```
$ python
Python 2.7.15 (default, Oct 23 2018, 01:23:26) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> import re
>>> cmd = 'free'
>>> mem_str = str(os.popen(cmd).readlines())
>>> print(mem_str)
['              total        used        free      shared  buff/cache   available\n', 'Mem:        8190340     2670172      385776         936     5134392     5215216\n', 'Swap:      35651500         256    35651244\n']
>>> mem = int(re.search(r'\d+', mem_str).group())
>>> print(mem)
8190340
>>> 
```

It seems Ok for 8GB RAM
